src/uart_arduino_nano.cpp

```diff
@@ -3,33 +3,112 @@
 #ifdef PLATFORM_ARDUINO_NANO
 
 #include "uart.h"
-#include <Arduino.h>
+#include <avr/io.h>
+#include <avr/interrupt.h>
 #include <stdarg.h>
+#include "ports.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+typedef struct
+{
+    volatile uint8_t busy;
+    volatile uint8_t numBytesToTransmit;
```

Comment:
to się da napisać zamiast na licznikach uint8_t na wskaźnikach i sizeof() - wtedy by było koszernie.

Reply:
Masz rację, niepotrzebnie chciałem skorzystać tutaj z bufora cyklicznego. Zrobiłem to teraz na wskaźnikach i wszystko się uprościło. Dzięki :)